rtsp_to_hls/hls_proxy_api/lib/connection/connection_manager.ex

```diff
@@ -0,0 +1,399 @@
+  end
+
+  @impl true
+  def handle_call(:close, _from, connection_status) do
+    {:disconnect, :close, connection_status}
+  end
+
+  @impl true
+  def handle_call({:reload, new_connection_status}, _from, connection_status) do
+    {:disconnect, :reload, Map.merge(connection_status, new_connection_status)}
+  end
+
+  @impl true
+  def handle_call(:status, _from, %{status: status} = connection_status) do
+    {:reply, status, connection_status}
+  end
+
+  @impl true
+  def handle_info(
+        {:DOWN, _ref, :process, pid, reason},
+        %ConnectionStatus{
+          rtsp_session: rtsp_session,
+          pipeline: pipeline,
+          keep_alive: keep_alive
+        } = connection_status
+      )
+      when reason != :normal do
+    Logger.warn(
+      "ConnectionManager_#{connection_status.stream.id}: Received DOWN message from #{
+        inspect(pid)
+      }"
+    )
+
+    Logger.warn(~s"""
+    ConnectionManager_#{connection_status.stream.id} processes:
+      RTSP session: #{inspect(rtsp_session)},
+      Membrane Pipeline: #{inspect(pipeline)},
+      RTSP keep alive: #{inspect(keep_alive)}
+    """)
+
+    case pid do
+      ^rtsp_session ->
+        Logger.error("ConnectionManager_#{connection_status.stream.id}: RTSP session crashed")
+
+      ^pipeline ->
+        Logger.error("ConnectionManager_#{connection_status.stream.id}: Pipeline crashed")
+
+      ^keep_alive ->
+        Logger.error(
+          "ConnectionManager_#{connection_status.stream.id}: Keep_alive process crashed"
+        )
+
+      process ->
+        Logger.error(
+          "ConnectionManager_#{connection_status.stream.id}: #{process} process crashed"
+        )
+    end
+
+    {:disconnect, :reload, connection_status}
+  end
+
+  @impl true
+  def handle_info(
+        {:DOWN, _ref, :process, _pid, reason},
+        connection_status
+      )
+      when reason == :normal do
+    {:noreply, connection_status}
+  end
+
+  @impl true
+  def handle_info({:EXIT, _from, reason}, connection_status) do
+    {:disconnect, {:error, reason}, connection_status}
+  end
+
+  defp start_rtsp_session(%ConnectionStatus{
+         rtsp_session: rtsp_session,
+         stream: %Stream{stream_url: stream_url, id: id}
+       }) do
+
+    if is_nil(rtsp_session) do
+      case RTSP.start(stream_url) do
+        {:ok, session} ->
+          Process.monitor(session)
+          session
+
+        {:error, error} ->
+          Logger.warn("ConnectionManager_#{id}: Starting RTSP session failed - #{inspect(error)}")
+          nil
+      end
+    else
+      rtsp_session
+    end
+  end
+
+  defp get_rtsp_description(%ConnectionStatus{rtsp_session: rtsp_session} = connection_status) do
+    Logger.debug("ConnectionManager_#{connection_status.stream.id}: Setting up RTSP description")
+
+    case RTSP.describe(rtsp_session) do
+      {:ok, %{status: 200} = response} ->
+        attributes = get_video_attributes(response)
```

Comment:
I believe we shouldn't pass the whole HTTP response, since `response.body`, or even `response.body.sdp_media` would be enough (`get_video_attributes` should deal only with SDP, not the whole HTTP response)

Reply:
Done